pkg/server/admin.go

```diff
@@ -1397,6 +1397,43 @@ func (s *adminServer) checkReadinessForHealthCheck() error {
 		return status.Errorf(codes.Unavailable, "node is shutting down")
 	}
 
+	if !s.server.sqlServer.acceptingClients.Get() {
+		return status.Errorf(codes.Unavailable, "node is not accepting SQL clients")
+	}
+
+	// Also check the availability of the SQL service.
+	// We do need a timeout on the query, to ensure the health probe
+	// returns a result within a reasonable timeframe, and that
+	// unavailable ranges translate to a health failure.
+	timeoutCtx, cancel := context.WithTimeout(ctx, 1*time.Second)
+	defer cancel()
+	if err := s.server.RunLocalSQL(timeoutCtx, func(ctx context.Context, ie *sql.InternalExecutor) error {
```

Comment:
Don't we need some form of rate limiting here to ensure that this code doesn't trigger at a high rate? The health endpoint is the most accessible endpoint in the system.